### ISSUE TITLE
mongodb-agent pod throwing error  "panic: Failed to get current user: user: unknown userid 1000490000"

### ISSUE THREAD

**Issue by rakeshk121** [mongodb/mongodb-kubernetes-operator#285]
**What did you do to encounter the bug?**
```
git clone https://github.com/mongodb/mongodb-kubernetes-operator.git
cd mongodb-kubernetes-operator 
oc new-project mongodb
oc create -f deploy/crds/mongodb.com_mongodb_crd.yaml -n mongodb
oc create -f deploy/operator/role.yaml -n mongodb
oc create -f deploy/operator/role_binding.yaml -n mongodb
oc create -f deploy/operator/service_account.yaml -n mongodb 
oc apply -f deploy/openshift/operator_openshift.yaml -n mongodb
oc apply -f deploy/crds/mongodb.com_v1_mongodb_openshift_cr.yaml -n mongodb
```

**What did you expect?**
Operator pod up and running and mongodb replicaset pods up and running

**What happened instead?**
Operator pod is successfully running but mongodb replica set pods does not come up. Error is as follows
```
[kubenode@master mongodb-kubernetes-operator]$ oc get pods
NAME                                           READY     STATUS             RESTARTS   AGE
example-openshift-mongodb-0                    1/2       CrashLoopBackOff   4          2m
mongodb-kubernetes-operator-66bfcbcf44-9xvj7   1/1       Running            0          2m

[kubenode@master mongodb-kubernetes-operator]$ oc logs -f example-openshift-mongodb-0 -c  mongodb-agent
panic: Failed to get current user: user: unknown userid 1000510000

goroutine 1 [running]:
com.tengen/cm/util.init.3()
        /data/mci/2f46ec94982c5440960d2b2bf2b6ae15/mms-automation/build/go-dependencies/src/com.tengen/cm/util/user.go:14 +0xe5

```

**Operator Information**
 - Operator Version:                  0.3.0
 - MongoDB Image used          4.2.6

**Kubernetes Cluster Information**
```
[kubenode@master mongodb-kubernetes-operator]$ openshift version
openshift v3.11.0+62803d0-1

[kubenode@master mongodb-kubernetes-operator]$ kubectl version
Client Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0+d4cacc0", GitCommit:"d4cacc0", GitTreeState:"clean", BuildDate:"2018-10-15T09:45:30Z", GoVersion:"go1.10.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0+d4cacc0", GitCommit:"d4cacc0", GitTreeState:"clean", BuildDate:"2020-12-07T17:59:40Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
```

I have gone through all the issues raised in the github regarding this issue and i see that as a solution set the environmental variable `MANAGED_SECURITY_CONTEXT` to true in operator, mongodb and mongodb-agent. I have done this to all theese three but still facing the same issue.
Here is the confirmation that environmental variables are correctly set:
```
[kubenode@master mongodb-kubernetes-operator]$ oc set env statefulset.apps/example-openshift-mongodb --list
# statefulsets/example-openshift-mongodb, container mongodb-agent
AGENT_STATUS_FILEPATH=/var/log/mongodb-mms-automation/healthstatus/agent-health-status.json
AUTOMATION_CONFIG_MAP=example-openshift-mongodb-config
HEADLESS_AGENT=true
MANAGED_SECURITY_CONTEXT=true
# POD_NAMESPACE from field path metadata.namespace
# statefulsets/example-openshift-mongodb, container mongod
AGENT_STATUS_FILEPATH=/healthstatus/agent-health-status.json
MANAGED_SECURITY_CONTEXT=true
[kubenode@master mongodb-kubernetes-operator]$ oc set env deployment.apps/mongodb-kubernetes-operator --list
# deployments/mongodb-kubernetes-operator, container mongodb-kubernetes-operator
# WATCH_NAMESPACE from field path metadata.namespace
# POD_NAME from field path metadata.name
MANAGED_SECURITY_CONTEXT=true
OPERATOR_NAME=mongodb-kubernetes-operator
AGENT_IMAGE=quay.io/mongodb/mongodb-agent:10.19.0.6562-1
VERSION_UPGRADE_HOOK_IMAGE=quay.io/mongodb/mongodb-kubernetes-operator-version-upgrade-post-start-hook:1.0.2
[kubenode@master mongodb-kubernetes-operator]$

```

**Comment by chatton** [comment]
Hi @rakeshk121 this looks like to be an issue that is effecting openshift specifically and not vanilla k8s, we believe that this should be resolved in the next release (0.5.2). 

This should be available early next week.

**Comment by bznein** [comment]
Hi @rakeshk121, 0.5.2 has now been released, can you try with the new version to see if the issue persists? Thanks
